lib/rubygems/version.rb

```diff
@@ -149,6 +149,15 @@
 # For the last example, single-digit versions are automatically extended with
 # a zero to give a sensible result.
 
+# Our code style opens classes directly without opening the intermediate
+# modules. This works because tha main entrypoint `rubygems.rb`, which defines
+# the root `Gem` module, is usually required first. But in this case we want to
+# allow using `Gem::Version` without loading the rest of rubygems, so we
+# explicit define the `Gem` placeholder module first.
```

Comment:
The `module Gem; end` will load all of RubyGems if there is `autoload :Gem, 'rubygems'` defined.
That can be detected with `if Object.autoload?(:Gem)`, but anyway I think there is nothing sensible we can do except loading the whole RubyGems on `require 'rubygems/version'` for that case.

This means, this approach does not work as uintended when `autoload :Gem, 'rubygems'` is used to load RubyGems lazily.

Reply:
Do you have any example of autoloading Gem lazily? When is that useful?

Reply:
TruffleRuby does it, because it's a huge gain on startup time: https://eregon.me/blog/2019/04/24/how-truffleruby-startup-became-faster-than-mri.html#autoloading-rubygems